src/NumericalAlgorithms/Spectral/SwshCollocation.hpp

```diff
@@ -0,0 +1,256 @@
+  Collocation() = default;
+  Collocation(const Collocation&) = default;
+  Collocation(Collocation&&) = default;
+  Collocation& operator=(Collocation&) = default;
+  Collocation& operator=(Collocation&&) = default;
+
+  /// retrieve the `sharp_geom_info*` stored. This should largely be used only
+  /// for passing to other libsharp functions. Otherwise, access elements
+  /// through iterator or access functions.
+  // clang-tidy seems to think the pointer will implicitly convert to the
+  // libsharp type. It does not.
+  sharp_geom_info* get_sharp_geom_info() const noexcept {
+    return *(geom_info_.get());  // NOLINT
+  }
+
+  /// Retrieve the \f$\theta\f$ value for a given index in a libsharp-compatible
+  /// array
+  double theta(size_t offset) const noexcept;
+  /// Retrieve the \f$\phi\f$ value for a given index in a libsharp-compatible
+  /// array
+  double phi(size_t offset) const noexcept;
+
+  constexpr size_t l_max() const noexcept { return l_max_; }
+
+  /// Compute the number of entries the libsharp-compatible data structure
+  /// should have
+  constexpr size_t size() const noexcept {
+    return (l_max_ + 1) * (2 * l_max_ + 1);
+  }
+
+  /// Get a bidirectional iterator to the start of the grid. `operator*` for
+  /// that iterator gives a
+  /// `std::pair<size_t /*offset*/ std::pair<double /*theta*/, double /*phi*/>`
+  const Collocation<Representation>::CollocationIterator begin() noexcept;
+  /// Get a bidirectional iterator to the end of the grid. `operator*` for
+  /// that iterator gives a
+  /// `std::pair<size_t /*offset*/ std::pair<double /*theta*/, double /*phi*/>`
+  const Collocation<Representation>::CollocationIterator end() noexcept;
+
+ private:
+  // an extra pointer layer is required for the peculiar way that libsharp
+  // constructs these values.
+  std::unique_ptr<sharp_geom_info*, detail::DestroySharpGeometry> geom_info_;
+  size_t l_max_ = 0;
+};
+
+bool operator==(typename Collocation<ComplexRepresentation::RealsThenImags>::
+                    CollocationIterator lhs,
+                typename Collocation<ComplexRepresentation::RealsThenImags>::
+                    CollocationIterator rhs) noexcept;
+bool operator!=(typename Collocation<ComplexRepresentation::RealsThenImags>::
+                    CollocationIterator lhs,
+                typename Collocation<ComplexRepresentation::RealsThenImags>::
+                    CollocationIterator rhs) noexcept;
+
+bool operator==(
+    typename Collocation<
+        ComplexRepresentation::Interleaved>::CollocationIterator lhs,
+    typename Collocation<
+        ComplexRepresentation::Interleaved>::CollocationIterator rhs) noexcept;
+bool operator!=(
+    typename Collocation<
+        ComplexRepresentation::Interleaved>::CollocationIterator lhs,
+    typename Collocation<
+        ComplexRepresentation::Interleaved>::CollocationIterator rhs) noexcept;
+
+namespace detail {
+// This function is threadsafe because of the very limited use of the static
```

Comment:
We don't comment on our use of `static` anywhere else. `static` is guaranteed to be thread safe by the language

Reply:
It is true that this is a language feature, but for this point we had previously discussed that because there are some pains gone through to ensure that the static cached object is only altered in the initializer, it should be mentioned that that use (and not uses which alter the static in subsequent lines) is guaranteed threadsafe. 
If you still want the comment gone, I'll remove it. I just wanted to confirm.

Reply:
Okay, I see what you're trying to say with the comment and to me it sounds like you're explaining that `static` is thread-safe at construction. How about:
```cpp
// We use a `std::index_sequence` to loop over the functions `precomputed_static_collocation_impl`
// we would like to only compute and store a specific `l_max` that is requested, not all values up to
// `MAXIMUM_L_MAX`. This cannot be done in the more traditional way of either having a `std::vector`
// or `std::undordered_map` that is `static` because only the construction phase is guaranteed to be
// thread-safe, not the access. This would mean were we to use a `std::vector` or `std::unordered_map`
// we would need to compute all values up to `MAXIMUM_L_MAX` immediately. By doing the loop over
// `Is` we are able to generate `sizeof...(Is)` different static objects and thus each one can be
// constructed only when it is needed and in a thread-safe manner.
```
?